Service/Request/Visitor/RemoveNullFieldsRequestVisitor.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIClientBundle\Service\Request\Visitor;
+
+use Http\Message\StreamFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Class RemoveNullFieldsRequestVisitor.
+ */
+class RemoveNullFieldsRequestVisitor implements RequestVisitorInterface
+{
+    /**
+     * @var EndpointConfiguration[]
+     */
+    private $endpoints;
+
+    /**
+     * @var StreamFactory
+     */
+    private $factory;
+
+    /**
+     * RemoveNullFieldsRequestVisitor constructor.
+     *
+     * @param StreamFactory $factory
+     * @param EndpointConfiguration[] $endpoints
+     */
+    public function __construct(
+        StreamFactory $factory,
+        array $endpoints
+    ) {
+        $this->factory = $factory;
+        $this->endpoints = $endpoints;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visit(RequestInterface $request): RequestInterface
+    {
+        if ($this->isCompatible($request)) {
+            $request = $request->withBody($this->cleanBody($request->getBody()));
+        }
+
+        return $request;
+    }
+
+    /**
+     * @return bool
+     */
+    private function isCompatible(RequestInterface $request): bool
+    {
+        foreach ($this->endpoints as $endpoint) {
+            if (
+                $request->getMethod() === $endpoint->getMethod()
+                && preg_match($endpoint->getPathRegexp(), $request->getUri()->getPath())
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return StreamInterface
+     */
+    private function cleanBody(StreamInterface $requestBody): StreamInterface
+    {
+        $data = json_decode($requestBody->getContents(), true);
```

Comment:
I'm not sure that we can do `json_decode`/`json_encode` here as instead of using `auto1.api.request.serializer` as the original application does.

It's important to encode/decode in the same way using the same serializer(with same configuration) as original query.